ee/secureenclavesigner/secureenclavesigner_darwin.go

```diff
@@ -0,0 +1,295 @@
+//go:build darwin
+// +build darwin
+
+package secureenclavesigner
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+	"os/user"
+	"strings"
+	"sync"
+
+	"github.com/kolide/krypto/pkg/echelper"
+	"github.com/kolide/launcher/ee/agent/types"
+	"github.com/kolide/launcher/ee/allowedcmd"
+	"github.com/kolide/launcher/ee/consoleuser"
+	"github.com/kolide/launcher/pkg/traces"
+)
+
+const (
+	CreateKeyCmd     = "create-key"
+	PublicEccDataKey = "publicEccData"
+)
+
+type opt func(*secureEnclaveSigner)
+
+type secureEnclaveSigner struct {
+	uidPubKeyMap         map[string]*ecdsa.PublicKey
+	pathToLauncherBinary string
+	store                types.GetterSetterDeleter
+	slogger              *slog.Logger
+	mux                  *sync.Mutex
+}
+
+func New(ctx context.Context, slogger *slog.Logger, store types.GetterSetterDeleter, opts ...opt) (*secureEnclaveSigner, error) {
+	ctx, span := traces.StartSpan(ctx)
+	defer span.End()
+
+	ses := &secureEnclaveSigner{
+		uidPubKeyMap: make(map[string]*ecdsa.PublicKey),
+		store:        store,
+		slogger:      slogger.With("component", "secureenclavesigner"),
+		mux:          &sync.Mutex{},
+	}
+
+	data, err := store.Get([]byte(PublicEccDataKey))
+	if err != nil {
+		traces.SetError(span, fmt.Errorf("getting public ecc data from store: %w", err))
+		return nil, fmt.Errorf("getting public ecc data from store: %w", err)
+	}
+
+	if data != nil {
```

Comment:
What happens if it's nil? That branch isn't populated. Maybe early return?

Reply:
I don't we would gain anything with an early return. With the intention of having key available as soon as possible, were making a call to `currentConsoleUserKey` that will get/create keys near bottom of func. We would have to add another line calling this if we early returned.

Reply:
Hrm. That's reasonable. But I think points to some lurking need to refactor. Maybe the quick path is to leave a comment, and call it good enough